### PR TITLE
feat(FeildsEqual,PathMap): add utility methods for comparing datasets

### PR DESCRIPTION
**Pull request by b5**
I think we should start shifting the "dynamic" methods for working with datasets down into the dataset package itself. Most of this is defined in [`github.com/qri-io/qri/base/component`](https://github.com/qri-io/qri/blob/master/base/component/component.go).

We should start selectively pulling the the utilitarian methods that express datasets & components using dynamic types & interfaces down into the dataset package, but adding _lots_ of tests in the process. This PR is a start to that, adding a `PathMap` method to dataset, and "runtime comparison" method to components that are inlined at save time.

### FieldsEqual methods
I'm not sure how to name this, but here's the general problem this method solves: 

you have two components that are in different states of persistence that need to be checked for equality. For example one "viz" component is saved to `/ipfs/`, and the other "viz" is not yet saved, or one "viz" component is saved to `/s3/` and the other is saved to `/ipfs/`. `FieldsEqual` checks if they're the same.

Basically this means "check everything but the component `Path` field and open script files. It comes with the caveat that fields that reference _other_ paths like `ScriptPath` are checked for equality (because two components that reference different script files are different.

### CODE REVIEW

File: readme_test.go [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+		{nil, nil, nil},
+		{&Readme{}, readme1, readme1},
+		{&Readme{
+			Format:     "bar",
+			Qri:        KindReadme.String(),
+			ScriptPath: "replace me",
+		},
+			readme2, readme2},
+		// {&Readme{
+		// 	Format:     "bar",
```
Comment (Arqu) [inline readme_test.go:77]: Dead code? or maybe make another negative test with this one

**Review by Arqu — APPROVED** [review]
LGTM, this will be handy on the change report for the "meta" status on components

**Comment by Arqu** [comment]
Regarding `FieldsEqual` maybe `ShallowCompare` or just `Compare` as it's a content based comparison

**Comment by b5** [comment]
Love it, renamed to `ShallowCompare`
